web-api/src/personalization/routes.rs

```diff
@@ -497,26 +526,34 @@ impl UnvalidatedSemanticSearchRequest {
             include_properties,
             filter,
         } = self;
-        let document = document.validate()?;
         let semantic_search_config: &SemanticSearchConfig = config.as_ref();
-        let count = validate_count(
+        let tenants_config: &tenants::Config = config.as_ref();
+
+        let document = document.validate()?;
+        let count = count.unwrap_or(semantic_search_config.default_number_documents);
+        dev.validate(tenants_config.enable_dev, count)?;
+        let num_candidates = dev
+            .max_number_candidates
+            .unwrap_or(semantic_search_config.max_number_candidates);
+        validate_count(
```

Comment:
:wrench:  We now have the  `count < max_number_candidates` validation twice.

Once as part of the dev options and once outside of the dev options, this is prone to introducing confusions and bugs when changing things.

Given that `dev.validate`  validates if the dev options are valid it should not check `count`  I think. 

Through it could error if `dev.max_number_candidates` has some absurd value, like `0` or some very very high number (if we want to do the later depends on weather we will give dev options to anyone but Dazzle or if it was just a super rare exception IMHO).